### ISSUE TITLE
CVE-2022-1471 not detected

### ISSUE THREAD

**Issue by blaargh** [aquasecurity/trivy#3303]
## Description
Trivy does not find recent CVEs in snakeyaml.
https://nvd.nist.gov/vuln/detail/CVE-2022-1471
https://github.com/advisories/GHSA-mjmj-j48q-9wg2

According to Trivy's datasources, two of the three available for Java packages (NVD and GitHub Advisory Database (Maven)) have this vulnerability listed.

## What did you expect to happen?
That the CVE is found
## What happened instead?
It wasn't

## Output of run with `-debug`:

```
dev@ide:~$ trivy -d rootfs tmp.jar
2022-12-14T12:25:25.844+0100    DEBUG   Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2022-12-14T12:25:25.846+0100    DEBUG   cache dir:  /home/dev/.cache/trivy
2022-12-14T12:25:25.847+0100    DEBUG   DB update was skipped because the local DB is the latest
2022-12-14T12:25:25.847+0100    DEBUG   DB Schema: 2, UpdatedAt: 2022-12-14 06:06:58.279798833 +0000 UTC, NextUpdate: 2022-12-14 12:06:58.279798333 +0000 UTC, DownloadedAt: 2022-12-14 10:30:41.483970433 +0000 UTC
2022-12-14T12:25:25.848+0100    INFO    Vulnerability scanning is enabled
2022-12-14T12:25:25.848+0100    DEBUG   Vulnerability type:  [os library]
2022-12-14T12:25:25.849+0100    INFO    Secret scanning is enabled
2022-12-14T12:25:25.850+0100    INFO    If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2022-12-14T12:25:25.850+0100    INFO    Please see also https://aquasecurity.github.io/trivy/v0.35/docs/secret/scanning/#recommendation for faster secret detection
2022-12-14T12:25:25.851+0100    DEBUG   No secret config detected: trivy-secret.yaml
2022-12-14T12:25:25.852+0100    DEBUG   Walk the file tree rooted at 'tmp.jar' in parallel
2022-12-14T12:25:25.852+0100    DEBUG   Parsing Java artifacts...       {"file": "tmp.jar"}
2022-12-14T12:25:26.697+0100    DEBUG   No such POM in the central repositories {"file": "tmp.jar"}
2022-12-14T12:25:26.701+0100    DEBUG   OS is not detected.
2022-12-14T12:25:26.701+0100    DEBUG   Detected OS: unknown
2022-12-14T12:25:26.701+0100    INFO    Number of language-specific files: 1
2022-12-14T12:25:26.701+0100    INFO    Detecting jar vulnerabilities...
2022-12-14T12:25:26.701+0100    DEBUG   Detecting library vulnerabilities, type: jar, path: 
2022-12-14T12:25:26.708+0100    INFO    Table result includes only package filenames. Use '--format json' option to get the full path to the package file.

```

## Output of `trivy -v`:

```
dev@ide:~$ trivy -v
Version: 0.35.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-12-14 06:06:58.279798833 +0000 UTC
  NextUpdate: 2022-12-14 12:06:58.279798333 +0000 UTC
  DownloadedAt: 2022-12-14 10:30:41.483970433 +0000 UTC

```

## Additional details (base image name, container registry info...):
The jar file used has snakeyaml version 1.33 installed
```
dev@ide:~$ jar tf tmp.jar | grep snakeyaml
com/fasterxml/jackson/dataformat/yaml/snakeyaml/
com/fasterxml/jackson/dataformat/yaml/snakeyaml/error/
com/fasterxml/jackson/dataformat/yaml/snakeyaml/error/Mark.class
com/fasterxml/jackson/dataformat/yaml/snakeyaml/error/MarkedYAMLException.class
com/fasterxml/jackson/dataformat/yaml/snakeyaml/error/YAMLException.class
com/fasterxml/jackson/dataformat/yaml/snakeyaml/error/package-info.class
META-INF/maven/org.yaml/snakeyaml/
META-INF/maven/org.yaml/snakeyaml/pom.properties
META-INF/maven/org.yaml/snakeyaml/pom.xml
org/yaml/snakeyaml/
org/yaml/snakeyaml/DumperOptions$FlowStyle.class
org/yaml/snakeyaml/DumperOptions$LineBreak.class
org/yaml/snakeyaml/DumperOptions$NonPrintableStyle.class
org/yaml/snakeyaml/DumperOptions$ScalarStyle.class
org/yaml/snakeyaml/DumperOptions$Version.class
[..]
```

**Comment by DmitriyLewen** [comment]
Hello @blaargh 
Thanks for your report!

Affected version for `CVE-2022-1471` ==  `v1.30`.
Take a look:
[NVD](https://nvd.nist.gov/vuln/detail/CVE-2022-1471)
[Google security research](https://github.com/google/security-research/security/advisories/GHSA-mjmj-j48q-9wg2) 

But looks like [GitHub database] contains mistake(affected versions - <= 1.30). We will write to them to fix this.

Regards, Dmitriy

**Comment by blaargh** [comment]
Thanks for the info. It's kinda weird that AWS Inspector reported this as critical when it doesn't even match our libraries version. Good to know that Trivy actually worked better in that case.

**Comment by snieguu** [comment]
> But looks like [GitHub database] contains mistake(affected versions - <= 1.30). We will write to them to fix this.
@DmitriyLewen do You have a response from Github?
It is not a mistake, the vulnerability is fixed but in snakyaml >=2.0
Reference:
https://bitbucket.org/snakeyaml/snakeyaml/wiki/Changes

**Comment by DmitriyLewen** [comment]
Hello @snieguu 
You are right. 
GitHub answered me about version range - https://github.com/github/advisory-database/pull/1371#issuecomment-1361837668.
GitLab also changed affected versions.

Trivy is currently finding CVE-2022-1471 for `snakeyaml < 2.0`.
